### PR TITLE
DEP: linalg: improve deprecation of private modules named without underscore prefix

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue

toward https://github.com/scipy/scipy/issues/18279
#### What does this implement/fix?

Currently some of the deprecation warnings for private modules named without underscore prefix are misleading. For example:
Before this pr:
```
In [2]: from scipy.linalg.basic import levinson
<ipython-input-2-6918a53fe48c>:1: DeprecationWarning: Please use `levinson` from the `scipy.linalg` namespace, the `scipy.linalg.basic` namespace is deprecated.
  from scipy.linalg.basic import levinson

In [3]: from scipy.linalg import levinson
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[3], line 1
----> 1 from scipy.linalg import levinson

ImportError: cannot import name 'levinson' from 'scipy.linalg' (/home/jakeb/development/scipy/build-install/lib/python3.10/site-packages/scipy/linalg/__init__.py)
```
After:
```
In [5]: from scipy.linalg.basic import levinson
<ipython-input-5-6918a53fe48c>:1: DeprecationWarning: `scipy.linalg.basic.levinson` is deprecated along with the `scipy.linalg.basic` namespace. `scipy.linalg.basic.levinson` will be removed in SciPy 1.13.0, and the `scipy.linalg.basic` namespace will be removed in SciPy 2.0.0.
```
This also reduces code duplication and strengthens testing.

### CODE REVIEW

File: scipy/linalg/flinalg.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ def __getattr__(name):
     if name not in __all__:
         raise AttributeError(
             "scipy.linalg.flinalg is deprecated and has no attribute "
-            f"{name}. Try looking in scipy.linalg instead.")
+            f"{name}.")
```
Comment (j-bowhay) [inline scipy/linalg/flinalg.py:17]: Wasn't sure about the best course of action here. Everything in the `__all__` of this module doesn't appear in `scipy.linalg`. This means if `_sub_module_deprecation` was used then they would all be stated for removal from `__all__` in 1.13.0 but then there would be nothing left in the module. However, the module isn't due to be removed until 2.0.0 hence why I just modified the message directly.

**Review by ilayn — APPROVED** [review]

File: scipy/linalg/flinalg.py (hunk 1 above)
Comment (h-vetinari) [reply scipy/linalg/flinalg.py:17]: I find it strange that we're keeping around an empty module, but the warning has already promised that'd we'd only remove it for 2.0.0, so 🤷

**Review by h-vetinari — APPROVED** [review]
Thanks! :)

File: scipy/linalg/flinalg.py (hunk 1 above)
Comment (ilayn) [reply scipy/linalg/flinalg.py:17]: I guess nobody anticipated that we would touch Fortran code. But I think that promise is void now.

**Comment by j-bowhay** [comment]
Everyone happy if we hit the button?

**Comment by h-vetinari** [comment]
Actually, this now needs to take into account the changes from https://github.com/scipy/scipy/pull/19067, e.g. we need to remove the `np.cast` aliases:
```
E   ImportError: cannot import name 'cast' from 'numpy'
```

File: scipy/linalg/flinalg.py (hunk 1 above)
Comment (h-vetinari) [reply scipy/linalg/flinalg.py:17]: See https://github.com/scipy/scipy/issues/19072

**Comment by j-bowhay** [comment]
updated

**Comment by h-vetinari** [comment]
Thanks @j-bowhay! :)

**Pull request merged**